src/v/config/configuration.h

```diff
@@ -110,7 +110,7 @@ struct configuration final : public config_store {
     property<bool> enable_transactions;
     property<uint32_t> abort_index_segment_size;
     // same as log.retention.ms in kafka
-    property<std::chrono::milliseconds> delete_retention_ms;
+    retention_property delete_retention_ms;
```

Comment:
can we make this more general like `disableable_property<T>` ?

Reply:
For retention -1 is a value to disable, for other properties it may be other value so I wouldn’t implement disableable_property just now